device_test.go

```diff
@@ -227,20 +217,54 @@ func (*failDetachLib) detachKernelDriver(h *libusbDevHandle, i uint8) error {
 
 func TestAutoDetachFailure(t *testing.T) {
 	t.Parallel()
-	fake := newFakeLibusb()
-	c := newContextWithImpl(&failDetachLib{fake})
-	defer c.Close()
-	dev, err := c.OpenDeviceWithVIDPID(0x8888, 0x0002)
-	if dev == nil {
-		t.Fatal("OpenDeviceWithVIDPID(0x8888, 0x0002): got nil device, need non-nil")
-	}
-	defer dev.Close()
+	dev, err := createFakeDevice(0x8888, 0x0002)
 	if err != nil {
-		t.Fatalf("OpenDeviceWithVIDPID(0x8888, 0x0002): %v", err)
+		t.Error(err)
 	}
 	dev.SetAutoDetach(true)
 	_, err = dev.Config(1)
 	if err == nil {
 		t.Fatalf("%s.Config(1) got nil, but want no nil because interface fails to detach", dev)
 	}
 }
+
+func TestActiveConfigNumFailure(t *testing.T) {
+	dev, err := createFakeDevice(0x8888, 0x0002)
+	if err != nil {
+		t.Error(err)
+	}
+
+	dev.handle = nil
```

Comment:
this looks to me like a wrong way to test this method. A better one would be to explicitly call dev.Close(). But given that this tests one line of code, and a very simple one at that, I don't think this test adds a lot of value.